### PR TITLE
Feature: Add cost tracking in jobs as metrics

### PR DESCRIPTION
**Pull request by mbeach-aws**
*Issue #, if available:*

*Description of changes:*
In the first four hybrid job notebooks, I added the Braket cost tracker as a jobs metric. This displays the total cost in near real-time on the Braket console or CloudWatch. The other notebooks use embedded simulators, so it is not applicable. This could be used as a way to limit costs for iterative jobs.  

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/hybrid_jobs/1_Quantum_machine_learning_in_Amazon_Braket_Hybrid_Jobs/qcbm/qcbm_job.py [hunk 1]
```diff
@@ -31,10 +30,16 @@ def main():
     print(f"Using device {device}")
 
     print("Starting circuit training...")
-    params = train_circuit(device, hyperparams, data)
+    params, cost_tracker = train_circuit(device, hyperparams, data)
```
Comment (shpface) [inline examples/hybrid_jobs/1_Quantum_machine_learning_in_Amazon_Braket_Hybrid_Jobs/qcbm/qcbm_job.py:33]: Rather than returning the cost tracker from `train_circuit`, we can have two separate cost trackers, one started in main to save the summary to the job result, and a second one confined to `train_circuit` for the purpose of logging.

File: examples/hybrid_jobs/1_Quantum_machine_learning_in_Amazon_Braket_Hybrid_Jobs/qcbm/qcbm_job.py [hunk 2]
```diff
@@ -62,10 +69,19 @@ def callback(x):
...
         loss = mmd_loss(qcbm.probabilities(x), data)
+
+        timestamp = time.time()
+
+        braket_cost = float(cost_tracker.simulator_tasks_cost() + cost_tracker.qpu_tasks_cost())
+
         log_metric(
-            metric_name="loss",
-            value=loss,
+            metric_name="braket_cost",
```
Comment (shpface) [inline examples/hybrid_jobs/1_Quantum_machine_learning_in_Amazon_Braket_Hybrid_Jobs/qcbm/qcbm_job.py:78]: Since the cost tracker is just an estimator of just Braket tasks, I think it would be good to rename braket_cost to something like braket_task_estimate or estimated_braket_task_cost to emphasize what the values represent and reduce opportunities for confusion.

Comment (mbeach-aws) [reply examples/hybrid_jobs/1_Quantum_machine_learning_in_Amazon_Braket_Hybrid_Jobs/qcbm/qcbm_job.py]: okay, updated to `braket_tasks_cost` to be consistent with the  `qpu_tasks_cost()` and `simulator_tasks_cost()` syntax

File: examples/hybrid_jobs/1_Quantum_machine_learning_in_Amazon_Braket_Hybrid_Jobs/qcbm/qcbm_job.py (hunk 1 above)
Comment (mbeach-aws) [reply examples/hybrid_jobs/1_Quantum_machine_learning_in_Amazon_Braket_Hybrid_Jobs/qcbm/qcbm_job.py]: okay, updated

**Review by shpface — APPROVED** [review]
